packages/backend/src/tenants/settings/service.test.ts

```diff
@@ -123,18 +125,76 @@ describe('TenantSetting Service', (): void => {
       expect(result[0].key).toEqual(initialOptions.setting[0].key)
       expect(result[0].value).toEqual(newValue)
     })
+
+    test.each`
+      key
+      ${TenantSettingKeys.EXCHANGE_RATES_URL.name}
+      ${TenantSettingKeys.WEBHOOK_MAX_RETRY.name}
+      ${TenantSettingKeys.WEBHOOK_TIMEOUT.name}
+      ${TenantSettingKeys.WEBHOOK_URL.name}
+      ${TenantSettingKeys.WALLET_ADDRESS_URL.name}
+    `(
+      'cannot use invalid setting value for $key',
+      async ({ key }): Promise<void> => {
+        const invalidSettingOption: CreateOptions = {
+          tenantId: tenant.id,
+          setting: [
+            {
+              key,
+              value: 'invalid_value'
+            }
+          ]
+        }
+
+        await expect(
+          tenantSettingService.create(invalidSettingOption)
+        ).resolves.toEqual(TenantSettingError.InvalidSetting)
+      }
+    )
+
+    test('cannot use invalid numeric values for numeric tenant settings', async (): Promise<void> => {
```

Comment:
should we add another one for URL validator?

Reply:
As in validating that URL-formatted tenant settings accept URLs, or that they don't accept non-URLs? The latter is tested in the test block before this one.

Reply:
> that they don't accept non-URLs
This one. And yes, looks like the test covers it 👍